cloudformation/replication.js

```diff
@@ -0,0 +1,122 @@
+module.exports = {
+	Parameters: {
+		"TrustedAWSPrinciples" : {
+			"Type" : "CommaDelimitedList",
+			"Description" : "List of AWS principles this stack trusts. (i.e. arn:aws:iam::<account_id>:root) Trusted accounts can assume the role of a bot on this stack and write to it."
+		},
+		"QueueReplicationDestinationLeoBotRoleARNs" : {
+			"Type" : "CommaDelimitedList",
+			"Description" : "List of LeoBotRole Arn's this stack will assume for replication. The AccountId and Stack of the first ARN become the default AccountId and Stack used when defining the QueueReplicationMapping."
+		},
+		"QueueReplicationMapping" : {
+			"Type" : "String",
+			"Default": "[]",
+			"Description" : "JSON Array of Objects and/or Strings the form [\"SOURCE_QUEUE_A\", {\"SOURCE_QUEUE_B\": { \"account\": \"DEST_ACCOUNT_ID\", \"stack\": \"DEST_STACK_NAME\", \"destination\":  \"DEST_QUEUE\"}}, {...}]. Omitting \"destination\" will default to the source queue name. Omitting \"account\" or \"stack\" will default to the first AccountId and Stack of the LeoBotRoleArn list. Simply listing a string will assume all defaults."
+		}
+	},
+	Conditions: {
+		IsTrustingAccount: {
+			"Fn::Not": [
+				{ "Fn::Equals" : [{ "Fn::Join" : [ ",", { "Ref" : "TrustedAWSPrinciples" } ] }, ""] }
+			]			
+		},
+		IsReplicatingStack: {
+			"Fn::And": [
+				{
+					"Fn::Not": [
+						{ "Fn::Equals" : [{ "Fn::Join" : [ ",", { "Ref" : "QueueReplicationDestinationLeoBotRoleARNs" } ] }, ""] }
+					]			
+				},
+				{
+					"Fn::Not": [
+						{ "Fn::Equals" : [{ "Ref" : "QueueReplicationMapping" }, "[]"] }
+					]			
+				}
+			]			
+		}
+	},
+	Resources: {
+		"SourceQueueReplicator": {
```

Comment:
Lets Add a comment here saying something like This is just the Condition for the bot. The remaining details will be add during the build.